src/main/java/seedu/address/model/person/Person.java

```diff
@@ -71,7 +86,8 @@ public boolean isSamePerson(Person otherPerson) {
 
         return otherPerson != null
                 && otherPerson.getName().equals(getName())
-                && (otherPerson.getPhone().equals(getPhone()) || otherPerson.getEmail().equals(getEmail()));
+                && (otherPerson.getPhone().equals(getPhone())
+                || otherPerson.getEmail().equals(getEmail()));
```

Comment:
Is this missing out the UID of otherPerson for comparing if both Persons are the same?

Reply:
This part checks for if you are trying to add a new person with the same name and phone or same name and email, so the UID can be different here.